### PR TITLE
add product upgrade flags to CR spec

### PR DESCRIPTION
**Pull request by tchughesiv**
`Spec.Upgrades.Patch` defaults to true

/assign @bmozaffa 

Signed-off-by: tchughesiv <tchughesiv@gmail.com>

### CODE REVIEW

File: deploy/examples/minor_upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: app.kiegroup.org/v1
+kind: KieApp
+metadata:
+  name: minor-upgrade
+spec:
+  environment: rhpam-trial
+  upgradeProductMinor: true
+  upgradeProductPatch: false
```
Comment (bmozaffa) [inline deploy/examples/minor_upgrade.yaml]: Why show this as false in this example?

File: deploy/crds/kieapp.crd.yaml [hunk 2]
```diff
@@ -605,3 +605,9 @@ spec:
                     replaceRole:
                       type: boolean
                       description: Whether to add to the current roles, or replace the current roles with the mapped ones. Replaces if set to true.
+            upgradeProductPatch:
+              type: boolean
+              description: Set false to disable automatic product patch version upgrades, it is enabled by default.
+            upgradeProductMinor:
+              type: boolean
+              description: Set true to enable automatic product minor version upgrades, it is disabled by default.
```
Comment (bmozaffa) [inline deploy/crds/kieapp.crd.yaml]: Let's do a struct called Upgrades, and then just minor & patch in there.

File: deploy/examples/minor_upgrade.yaml (hunk 1 above)
Comment (tchughesiv) [reply deploy/examples/minor_upgrade.yaml]: idk, good point

File: deploy/crds/kieapp.crd.yaml (hunk 2 above)
Comment (tchughesiv) [reply deploy/crds/kieapp.crd.yaml]: done

File: deploy/examples/minor_upgrade.yaml (hunk 1 above)
Comment (tchughesiv) [reply deploy/examples/minor_upgrade.yaml]: done

**Pull request merged**